Integrations/Perch/Perch.py

```diff
@@ -0,0 +1,494 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+''' IMPORTS '''
+
+import requests
+import json
+import collections
+import os
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+USERNAME = demisto.params().get('credentials').get('identifier')
+PASSWORD = demisto.params().get('credentials').get('password')
+API_KEY = demisto.params().get('api-key')
+FETCH_TIME = int(demisto.params().get('fetch_time', '7'))
+SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else \
+    demisto.params()['url']
+USE_SSL = not demisto.params().get('insecure', False)
+BASE_URL = SERVER + '/v1'
+if not demisto.params().get('proxy'):
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
+
+TLP_MAP = {
+    'WHITE': 0,
+    'GREEN': 1,
+    'AMBER': 2,
+    'RED': 3
+}
+
+CONFIDENCE_MAP = {
+    'LOW': 0,
+    'MEDIUM': 1,
+    'HIGH': 2
+}
+
+OBSERVABLE_TYPES_MAP = {
+    'IP': 0,
+    'Domain': 1,
+    'URL': 2,
+    'REGEX': 3,
+    'File Hash': 4
+}
+
+''' HELPER FUNCTIONS '''
+
+
+# Allows nested keys to be accessible
+def makehash():
+    return collections.defaultdict(makehash)
+
+
+def http_request(method, url_suffix, params=None, data=None, headers=None):
+    try:
+        res = requests.request(
+            method,
+            BASE_URL + url_suffix,
+            verify=USE_SSL,
+            params=params,
+            data=data,
+            headers=headers
+        )
+        if res.status_code == 403:
+            return_error('Connection forbidden. Please verify your API key is valid.')
+        elif res.status_code not in {200, 201}:
+            return_error('Error in API call to Perch Integration [%d] - %s' % (res.status_code, res.reason))
+
+    except requests.exceptions.ConnectionError as error:
+        return_error("Failed to establish a new connection: {error}".format(error=type(error)))
+
+    return res.json()
```

Comment:
please wrap the .json with a try and return error like "failed to parse response to error" and print the response contents.